_posts/2016-01-21-measuring-episerver-site-performance.md

```diff
@@ -0,0 +1,83 @@
+
+Eager as we are, let us fire up the test by pressing START, and view the results as Aggregate Report. In this test I used Episerver Alloy demo on localhost.
+
+[![JMeter report view](/img/measuring-episerver-site-performance/jmeter-report-view.jpg)](/img/measuring-episerver-site-performance/jmeter-report-view.jpg)
+
+The report is easy to interpret, just ensure that Error rate is always zero. 
+The most interesting column is Throughput as it's unit is requests per second, the answer we were looking for.
+Please note that only the html is downloaded; stylesheets, scripts and other embedded resources are ignored.
+
+While executing the test, the server CPU usage should be sky-high.
+
+![Maximize CPU usage](/img/measuring-episerver-site-performance/metalcrusher.jpg)
+
+If not, you can change the execution parameters to increase the load.
+By default, the project uses 15 threads for requests, that is 15 simultaneous requests running at a time. 
+Each thread will execute 100 requests, totalling to 1500 requests per each url.
+The defaults are changeable on _Test Plan_-node.
+
+![Change execution parameters](/img/measuring-episerver-site-performance/jmeter-your-best-nightmare.jpg)
+
+While we are at it, let us see the other options too.
+
+![Change execution parameters](/img/measuring-episerver-site-performance/jmeter-other-confs.jpg)
+
+You can enable grayed out nodes by left-clicking and choosing _Enable_ in the context menu.
+
+* **Cookie Manager**: Enables cookies. Threads will remember and use cookies like a browser. You can also add custom cookies that are used by all threads, for example, a session cookie for authentication.
+* **View Results in Table**: Shows details per request in a table. Not as verbose as View Results in Tree. 
+* **View Results in Tree**:Very detailed results per request. Contains http request and response details, thus making this very useful while debugging site access and errors.
+
+### Reliable site setup
+The most reliable results are acquired from an environment identical to production, unfortunately this is not always possible in development stage. 
+If you choose to use localhost, bear in mind that there is virtually no network delay or bandwidth limit, which will yield way too optimistic results. 
+Also, mind that running the site on Visual Studio with IIS Express will have unstable performance compare to real IIS. 
+I have often used CI environment when production or test environments were not available.
+
+![Real-life internet](/img/measuring-episerver-site-performance/real-life-internet.jpg)
+
+
+### Protip: Profiling the code
+JMeter may show you the performance rating, but profiling will tell you where is the hold-up. 
+Very useful for spotting performance issues in code, and to help avoiding premature optimization - just optimize what needs to be optimized. 
+JetBrain's [dotTrace](https://www.jetbrains.com/profiler/) is my tool of choice.
```

Comment:
It's "JetBrains", so JetBrains'. Or just remove the ' completely, that would work too